scripts/generate-react-exports.js

```diff
@@ -24,19 +24,18 @@ const baklavaReactFileParts = [];
 
 for (const module of customElementsModules) {
   const { declarations, path } = module;
-  const componentDeclaration = declarations.find(declaration => declaration.customElement === true);
-  const { events, name: componentName, tagName: fileName, jsDoc } = componentDeclaration;
+  const component = declarations.find(declaration => declaration.customElement === true);
+  const { name: componentName, tagName: fileName } = component;
 
-  const eventNames =
-    events?.reduce((acc, curr) => {
-      acc[getReactEventName(curr.name)] = curr.name;
-      return acc;
-    }, {}) || {};
+  const eventNames = (component.events || []).map(event => {
+    const reactEvent = getReactEventName(event.name);
+      const reactEventName = `${componentName}${reactEvent.split("onBl")[1]}`;
+      return `${reactEvent}: '${event.name}' as EventName<${reactEventName}>`;
```

Comment:
We generate a reference like `EventName<BlTextareaInpu>` but we don't import `BlTextareaInput` anywhere. I think that should cause some issues.

Reply:
It's the type of the component so it's referring to self, it shouldn't be a problem because we set BlTextareaInputType to BlTextareaInput above.

Reply:
I actually found a way to use the types without repeating the code, just pushed the changes.

Reply:
I may have misunderstood your question, the event types are being generated since #711.